### ISSUE TITLE
Create backend and frontend for connecting scratchnet instances

### ISSUE THREAD

**Issue by karel1980** [karel1980/scratchnet#25]
The connection process goes like this:
- Alice uses localhost:2000 to invite {Bob} for {chat}
- Bob receives the invitation (browser pops this up)
- Bob accepts the invitation (his manager starts comm.js and sends acceptance to Alice) 
- Alice receives the acceptance and starts comm.js

**Comment by marc-portier** [comment]
yep.
I think you will need a 5th step though: after the service is started on Alice's side, you will need to communicate the allocated port back to Bob.

And I suggest we use some "connection-setup-id" to handle this messaging between Bob and Alice, just to make sure they could be simultanuously setting up multiple connections with multiple connection-peers...

**Comment by karel1980** [comment]
Yes. I was aiming to put that in the response body of Alice's last step, but because of the async nature of the .listen() call we might need an additional call
